src/Microsoft.ServiceFabric.Actors.KVSToRCMigration/TargetMigrationOrchestrator.cs

```diff
@@ -588,19 +599,43 @@ private async Task<IMigrationPhaseWorkload> NextWorkloadRunnerAsync(PhaseResult
         {
             var endSN = await this.GetEndSequenceNumberAsync(cancellationToken);
             var delta = endSN - currentResult.EndSeqNum;
+
             if (currentResult.Phase == MigrationPhase.Catchup)
             {
-                if (delta < this.MigrationSettings.DowntimeThreshold
-                    && this.downtimeAllowed == true)
```

Comment:
why are we complicating the logic here with more if..else blocks. Cant' we update downtimeAllowed from MetaDataDictionary and keep the original logic.